LE Audio: Rename MICS to MICP and use role names

Thalley opened the pull request:
This PR basically renames the entire Microphone Input Control Profile (MICP) implementation. 

The server side is renamed to MICP Microphone Device (bt_micp_mic_dev), and the client side is renamed to MICP Microphone Controller (bt_micp_mic_ctlr). This generally add renamings such as mics->micp and then adding the role name, as well as renaming the public header and Kconfigs from MICS to MICP. 

This PR also removes the bt_micp_aics API, as the AICS instance pointer can be used directly with the AICS API instead with the same functionality, which cleans up the implementation significantly. 

fixes https://github.com/zephyrproject-rtos/zephyr/issues/45558

Comment for reviewers:
You can review however you like, but for your sake I would focus on the `micp.h` public header file, as well as the `Kconfig.micp` file, as everything else has just been simple renames based on the changes in those files (except for the removed `bt_mics_aics` functions).

Thalley committed:
Bluetooth: Audio: Rename mics.h the Kconfig to micp

Rename the files to match the name of the profile (MICP)
instead of the service (MICS).

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Rename MICS to MICP

Use the profile abbreviation, MICP, instead of the
service abbreviation, MICS, for the MICP implementation.

The server implementation will use MICP and the client
implementation will use MICP_CLIENT for now, as that will
be updated in a future commit.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Rename remaning MICS files to MICP roles

Rename mics => micp_mic_dev (MICP Microphone Device)
Rename mics_client -> micp_mic_ctlr (MICP Microphone Controller)

This does not change any lines in the code.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Rename bt_micp_aics functions

The AICS functionality is also possible via the bt_aics
API, and this was just a way to ensure that the AICS
instance was part of a MICS.

Removing makes MICP much simpler, and we can expect user
to know the context of a AICS.

This also fixes a few bad/missing guards.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Add separate API for MICP mic ctlr

Add a bt_micp_mic_ctlr API that is used only
for the MICP Microphone Controller (Client).

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Specify MICP Mic Dev API

Modify the existing bt_micp API for the
MICP Microphone Device to be more specific
by adding mic_dev as an infix. This follows
the naming scheme used for the MICP microphone
controller (mic_ctlr).

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Remove bt_micp argument for MICP mic_dev API

Since there is a split API now, the mic_dev does not need
to take a bt_micp pointer.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

Thalley committed:
Bluetooth: Audio: Rename struct bt_micp and cleanup

Rename struct bt_micp to bt_mic_ctlr as it is only
used ofr microphone controller now.

This also allow cleanup of the bt_micp struct, thus
removing the common internal micp_internal.h header file
and reducing the size of the implementation by not
using a union of the structs.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

asbjornsabo commented on subsys/bluetooth/shell/micp_mic_ctlr.c:
```diff
@@ -1,39 +1,46 @@
 {
 	if (err != 0) {
-		shell_error(ctx_shell, "MICS discover failed (%d)", err);
+		shell_error(ctx_shell, "MIC_CTLR discover failed (%d)", err);
```
Could be misunderstood.  It is the CTLR doing the discovery, but it is the DEV not being discovered.
Proposal: "MIC_CLTR failed to discover MIC_DEV".
(Before the change, the text was "MICS discover failed", and the MICS is the service.

Thalley replied:
Modified all places where the string "MIC_CLTR" was used to something less odd :)

asbjornsabo commented on tests/bluetooth/shell/testcase.yaml:
```diff
@@ -100,34 +100,34 @@ tests:
+      - CONFIG_BT_MICP_MIC_CTLR=n
     tags: bluetooth
-  bluetooth.shell.audio.no_mics_mics_client:
+  bluetooth.shell.audio.no_micp_micp_mic_ctlr:
```
"no_micp_mic_dev_micp_mic_ctlr"

asbjornsabo commented on tests/bluetooth/bsim_bt/bsim_test_audio/test_scripts/micp.sh:
```diff
@@ -24,10 +24,10 @@ BOARD="${BOARD:-nrf52_bsim}"
 cd ${BSIM_OUT_PATH}/bin
 
-printf "\n\n======== Running MICS Server Only (API) test =========\n\n"
+printf "\n\n======== Running MICP Server Only (API) test =========\n\n"
```
"MICP Server" -> "MICP Device"

asbjornsabo commented on tests/bluetooth/bsim_bt/bsim_test_audio/test_scripts/micp.sh:
```diff
@@ -37,13 +37,13 @@ for PROCESS_ID in $PROCESS_IDS; do
 done
 
-printf "\n\n======== Running MICS and MICS client test =========\n\n"
+printf "\n\n======== Running MICP and MICP client test =========\n\n"
```
"MICP device and MICP client"

asbjornsabo commented on tests/bluetooth/shell/testcase.yaml:
```diff
@@ -100,34 +100,34 @@ tests:
       - CONFIG_BT_VCS_CLIENT_MAX_VOCS_INST=0
     tags: bluetooth
-  bluetooth.shell.audio.no_mics:
+  bluetooth.shell.audio.no_micp:
```
"no_micp_mic_dev"

asbjornsabo commented on tests/bluetooth/bsim_bt/bsim_test_audio/test_scripts/micp.sh:
```diff
@@ -37,13 +37,13 @@ for PROCESS_ID in $PROCESS_IDS; do
 
 Execute ./bs_${BOARD}_tests_bluetooth_bsim_bt_bsim_test_audio_prj_conf \
-  -v=${VERBOSITY_LEVEL} -s=${SIMULATION_ID} -d=0 -testid=mics -rs=23
+  -v=${VERBOSITY_LEVEL} -s=${SIMULATION_ID} -d=0 -testid=micp -rs=23
```
"micp_mic_dev"

asbjornsabo reviewed (changes requested):
I think this renaming improves things - it makes things clearer.
Found some renamings to fix/improve - there may be more that I have not seen.

Thalley commented:
@asbjornsabo from the Naming Conventions RFC I think it was also suggested that "verb-noun" was recommended (https://github.com/zephyrproject-rtos/zephyr/pull/30892/files#r552095727), so I wonder if we should also change the API from e.g. `bt_micp_mic_dev_mute_get` to `bt_micp_mic_dev_get_mute` as part of these PRs as well?

Thalley commented on include/zephyr/bluetooth/audio/micp.h:
```diff
@@ -0,0 +1,281 @@
+ *
+ * @return 0 if success, errno on failure.
+ */
+int bt_micp_mic_dev_register(struct bt_micp_mic_dev_register_param *param);
```
> My default is to keep naming as is here.

Same

IMO `micp_dev` and `micp_ctlr` is "as bad" as just using `bap_source` and `bap_client` (especially the former, since "device" is very generic).

Thalley committed:
Bluetooth: Audio: Rename MICP mic_dev disable to mute disable

Made it more clear that the disable functionality is to
disable the mute functionality, but and not disabling
the microphone device itself.

Signed-off-by: Emil Gydesen <emil.gydesen@nordicsemi.no>

asbjornsabo commented:
> @asbjornsabo from the Naming Conventions RFC I think it was also suggested that "verb-noun" was recommended (https://github.com/zephyrproject-rtos/zephyr/pull/30892/files#r552095727), so I wonder if we should also change the API from e.g. `bt_micp_mic_dev_mute_get` to `bt_micp_mic_dev_get_mute` as part of these PRs as well?

Was there ever a clear conclusion to that?  At least the PR was not merged.
I can live with either, but suggest that we either wait with this change or (preferably) get some confirmation from the API group on what they prefer before doing the change.
(That change can be a separate commit, IMO.)

asbjornsabo reviewed (approved):
My comments have been addressed.
